api-examples/pretrain-lm-streaming.py

```diff
@@ -209,10 +210,24 @@ def train():
     start_epoch = 0
     if args.restart_from:
         model.load_state_dict(torch.load(args.restart_from))
-        start_epoch = int(args.restart_from.split("-")[-1].split(".")[0]) - 1
-        global_step = (start_epoch+1) * steps_per_epoch
+        vec = args.restart_from.split("-")
+
+        if args.restart_tt:
+            tick_type = args.restart_tt
+        else:
+            tick_type = vec[-2]
+        step_num = int(vec[-1].split(".")[0])
+        if tick_type == 'epoch':
+            start_epoch = step_num - 1
+            global_step = (start_epoch + 1) * steps_per_epoch
```

Comment:
In all the cases, `steps_per_epoch` are calculated as `number_of_train_samples // args.batch_size`, but i think it should be `number_of_train_samples // (args.batch_size*num_gpus)` since each worker only needs to run this many steps to finish a epoch. I cut my release and tested that `int(os.environ.get("WORLD_SIZE", 1))` does give the num_gpus in our cluster environment. I still think we should fix this so the scheduler works as expected.

Reply:
Can this be resolved?